core/src/test/java/com/linecorp/armeria/server/docs/JsonSchemaGeneratorTest.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright 2020 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.server.docs;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.util.List;
+
+import org.junit.jupiter.api.Test;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.google.common.collect.ImmutableList;
+
+class JsonSchemaGeneratorTest {
```

Comment:
Might consider adding more tests, those manual ones work fine but I am more interested in direct conversion from `proto` to `JSONSpec` and maybe even for `thrift`, `http` or `graphQL` (But I am not very confident on those).

Reply:
> I am more interested in direct conversion from proto to JSONSpec and maybe even for thrift, http or graphQL (But I am not very confident on those).

I was wondering why you prefer direct conversion. Is there loss of type information? Because of the flexibility of GraphQL, JSON schema might not handle GraphQL property but GraphQL provides a pluggable web IDE which we already support experimentally.

Reply:
Sorry I should have said to convert it inside the doc service plugin like `GrpcDocServicePlugin` for protos and `ThriftDocServicePlugin` or thrift etc.

`ServiceInfo` supports most of the things that I need but some important things are not accessible for example for enum types in protobuf I need to check whether `namedTypeInfoProvider` is an instance of `EnumDescriptor` or not. But I can't do that inside armeria-core package because its classpath doesn't contain `EnumDescriptor` (proto lib).

Reply:
Also I am more interested in `Proto` -> `JSONSpec` just for testing because it reflects the real-life usage better (mostly related to the `TypeInformation` being converted correctly).

Reply:
> for enum types in protobuf I need to check whether namedTypeInfoProvider is an instance of EnumDescriptor or not.

As far as I know, `EnumDescriptor` is converted into `EnumInfo`. Do you think `EnumInfo` does not provide enough information to convert proto into JSON schema?

> Also I am more interested in Proto -> JSONSpec just for testing 

It makes sense.

Reply:
@ikhoon I am trying to find where to find that `EnumInfo`. Maybe I should pass something else to my method to get the right type, seems like `FieldInfo` is not enough to infer that. I am kinda confused at that point.

Reply:
> seems like FieldInfo is not enough to infer that.

We might have to add `NamedTypeInfo` to `FieldInfo` for `MESSAGE` and `ENUM`.
I guess `NamedTypeInfo` could be removed from JSON serialization.
It seems out of scope for this PR. Let me create an issue and handle it separately.

Reply:
On second thought, it would be easy to add a `typeDescriptor` to `NamedTypeInfo`.
For example, we can change the following example into:
```java
new EnumInfo(..., enumDescriptor);
```
https://github.com/line/armeria/blob/8ed438da36ba4e0f7d111c00f748b3fcc2906e2c/protobuf/src/main/java/com/linecorp/armeria/server/protobuf/ProtobufNamedTypeInfoProvider.java#L226-L232

Reply:
Resolving this because I am using the new typed signatures.